File: tests/fixtures/langs/sample.pas
unit Sample;

interface

type
  TWidget = class
    Width: Integer;
    function Render: Integer;
  end;

function ComputeTotal(const Items: array of Integer): Integer;

implementation

function TWidget.Render: Integer;
begin
  Result := Width * 2;
end;

function ComputeTotal(const Items: array of Integer): Integer;
var I: Integer;
begin
  Result := 0;
  for I := Low(Items) to High(Items) do Result := Result + Items[I];
end;

end.

File: tests/fixtures/langs/sample.f90
module sample
  implicit none
contains
  function compute_total(items) result(t)
    integer, intent(in) :: items(:)
    integer :: t
    t = sum(items)
  end function compute_total

  subroutine render(width)
    integer, intent(in) :: width
    print *, repeat(' ', width)
  end subroutine render
end module sample
